### PR TITLE
Remove required fields from growable tables

### LINKED ISSUE #3164: Principal occupation doesn't need to be required
In Employment history: http://localhost:3000/education/apply-for-education-benefits/application/employment-history/employment-information

### PR DESCRIPTION
**Pull request by jbalboni**
Closes #3164.

This removes the required fields from all the GrowableTable rows, excluding military service. It also fixes a bug where ROTC history should have been requiring year of commission but wasn't.

screen shot 2016-09-23 at 11 45 22 am
screen shot 2016-09-23 at 11 45 50 am
screen shot 2016-09-23 at 11 46 20 am
screen shot 2016-09-23 at 11 47 06 am

### CODE REVIEW

**Comment by jbalboni** [comment]
@akainic Can you take a look?

File: src/js/edu-benefits/components/military-history/RotcScholarship.jsx [hunk 1]
```diff
@@ -2,15 +2,9 @@ import React from 'react';
...
+import { isValidYear, validateIfDirty, isValidMonetaryValue, isBlank } from '../../utils/validations';
 
-function isValidRequiredAmount(val) {
-  return isNotBlank(val) && isValidMonetaryValue(val);
-}
-
-function isValidRequiredYear(val) {
-  return isNotBlank(val) && isValidYear(val);
-}
+const validateIfNotBlank = (validator) => (val) => isBlank(val) || validator(val);
```
Comment (akainic) [inline src/js/edu-benefits/components/military-history/RotcScholarship.jsx]: should we just move this into the validations.js file to import here, if it doesn't already exist there?

**Review by akainic — APPROVED** [review]
LGTM with one nit!

File: src/js/edu-benefits/components/military-history/RotcScholarship.jsx (hunk 1 above)
Comment (jbalboni) [inline src/js/edu-benefits/components/military-history/RotcScholarship.jsx]: Yep, I'll move it.

**Pull request merged**